tools/src/component-generator.ts

```diff
@@ -252,7 +252,8 @@ const renderImports: (model: {
     `import { PropTypes } from "prop-types";` + `\n` +
 `<#?#>` +
 
-`import { <#= it.baseComponentName #> as BaseComponent } from "<#= it.baseComponentPath #>";` + `\n` +
+`import { <#= it.baseComponentName #> as BaseComponent` +
+    `<#? it.hasExtraOptions #>, IHtmlOptions<#?#> } from "<#= it.baseComponentPath #>";` + `\n` +
```

Comment:
To keep templates simply readable, all conditional rendering constructions are currently split into several lines (e.g. see line 251).

So
```
`import { <#= it.baseComponentName #> as BaseComponent` +

`<#? it.hasExtraOptions #>` +
    `, IHtmlOptions` +
`<#?#>` +

` } from "<#= it.baseComponentPath #>";` + `\n` +
```